### ISSUE TITLE
Hibernate schema update does not create makumba SET tables correctly

### ISSUE THREAD

**Issue by ghost** [makumba/makumba#581]
**Reported by @cristianbogdan on 1 May 2008 01:46 UTC**
I added in a type called projman.Project a set field

signExecution= set projman.Company

at the next schema update, hibernate created the following table:

projman_Project__signExecution_ | CREATE TABLE `projman_Project__signExecution_` (
  `Project_` int(11) NOT NULL,
  `Company_` int(11) NOT NULL,
  KEY `FKACC2455B4C5273CE` (`Company_`),
  KEY `FKACC2455BE974474E` (`Project_`),
  CONSTRAINT `FKACC2455BE974474E` FOREIGN KEY (`Project_`) REFERENCES `projman_project_` (`Project_`),
  CONSTRAINT `FKACC2455B4C5273CE` FOREIGN KEY (`Company_`) REFERENCES `projman_company_` (`Company_`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |

my project didn't work (with a strange NPE in TableManager.insertRecordImpl!!!)  so i enabled alter#=true. that dropped the hibernate indexes (!) and after the subsequent setting of alter to false, and another hibernate schema update, i got

projman_Project__signExecution_ | CREATE TABLE `projman_Project__signExecution_` (
  `Project_` int(11) NOT NULL,
  `Company_` int(11) NOT NULL,
  `signExecution_` int(11) default NULL,
  `TS_modify_` timestamp NOT NULL default CURRENT_TIMESTAMP on update CURRENT_TIMESTAMP,
  `TS_create_` timestamp NOT NULL default '0000-00-00 00:00:00',
  UNIQUE KEY `signExecution_` (`signExecution_`),
  KEY `TS_modify_` (`TS_modify_`),
  KEY `TS_create_` (`TS_create_`),
  KEY `Project_` (`Project_`),
  KEY `Company_` (`Company_`),
  CONSTRAINT `FKACC2455B4C5273CE` FOREIGN KEY (`Company_`) REFERENCES `projman_company_` (`Company_`),
  CONSTRAINT `FKACC2455BE974474E` FOREIGN KEY (`Project_`) REFERENCES `projman_project_` (`Project_`)
) ENGINE=InnoDB 

in short, hibernate doesn't create the primary key, TS_create and TS_modify, and (less importantly) it doesn't create indexes. the missing fields make it impossible to operate for the makumba db logic (e.g. for set updates).

i know that our hibernate db operation tests (the equivalent of the makumba db operations) pass, so i wonder how are the missing fields updated by hibernate since it doesn't know about them...

Migrated-From: http://trac.makumba.org/ticket/1035

**Comment by ghost** [comment]
**Modified by @manuelbernhardt on 28 Jun 2010 07:22 UTC**

**Comment by ghost** [comment]
**Modified by @manuelbernhardt on 28 Jun 2010 07:36 UTC**
